component-samples/scripts/keys_gen.sh

```diff
@@ -0,0 +1,247 @@
+        openssl req -x509 -key $CERTS_PATH/${1}_rsa2048_private.key -out $CERTS_PATH/${1}_rsa2048_cert.pem -days 3650 -batch
+
+        # EC256 Keys Generations
+        openssl ecparam -name prime256v1 -genkey -noout -out $CERTS_PATH/${1}_ec256_private.key
+        openssl ec -in $CERTS_PATH/${1}_ec256_private.key -pubout -out $CERTS_PATH/${1}_ec256_public.key
+        openssl req -x509 -key $CERTS_PATH/${1}_ec256_private.key -out $CERTS_PATH/${1}_ec256_cert.pem -days 3650 -batch
+
+        # EC384 Keys Generations
+        openssl ecparam -name secp384r1 -genkey -noout -out $CERTS_PATH/${1}_ec384_private.key
+        openssl ec -in $CERTS_PATH/${1}_ec384_private.key -pubout -out $CERTS_PATH/${1}_ec384_public.key
+        openssl req -x509 -key $CERTS_PATH/${1}_ec384_private.key -out $CERTS_PATH/${1}_ec384_cert.pem -days 3650 -batch
+
+        # Export RSA2048, EC256, and EC384 keys to p12
+        openssl pkcs12 -export -in $CERTS_PATH/${1}_rsa2048_cert.pem -inkey $CERTS_PATH/${1}_rsa2048_private.key \
+                -name ${1}_rsa2048 -out $CERTS_PATH/${1}_rsa2048.p12 -password pass:${PASS_KEY}
+        openssl pkcs12 -export -in $CERTS_PATH/${1}_ec256_cert.pem -inkey $CERTS_PATH/${1}_ec256_private.key \
+                -name ${1}_ecdsa_256 -out $CERTS_PATH/${1}_ec256.p12 -password pass:${PASS_KEY}
+        openssl pkcs12 -export -in $CERTS_PATH/${1}_ec384_cert.pem -inkey $CERTS_PATH/${1}_ec384_private.key \
+                -name ${1}_ecdsa_384 -out $CERTS_PATH/${1}_ec384.p12 -password pass:${PASS_KEY}
+
+        # Import rsa2048, ec256, and ec384 p12 keystores to one p12 keystore
+        keytool -importkeystore -srckeystore $CERTS_PATH/${1}_rsa2048.p12 -srcstorepass ${PASS_KEY} \
+                -destkeystore $CERTS_PATH/${1}_keystore.p12 -deststorepass ${PASS_KEY} -deststoretype pkcs12 -noprompt
+        keytool -importkeystore -srckeystore $CERTS_PATH/${1}_ec256.p12 -srcstorepass ${PASS_KEY} \
+                -destkeystore $CERTS_PATH/${1}_keystore.p12 -deststorepass ${PASS_KEY} -deststoretype pkcs12 -noprompt
+        keytool -importkeystore -srckeystore $CERTS_PATH/${1}_ec384.p12 -srcstorepass ${PASS_KEY} \
+                -destkeystore $CERTS_PATH/${1}_keystore.p12 -deststorepass ${PASS_KEY} -deststoretype pkcs12 -noprompt
+
+        # Verify the keys from p12 keystore
+        keytool -list -keystore $CERTS_PATH/${1}_keystore.p12 -storepass ${PASS_KEY}
+
+        # Creating a keys.pem files
+        cat $CERTS_PATH/${1}_ec256_public.key | tee $CERTS_PATH/${1}_pub_keys.pem > /dev/null
+        cat $CERTS_PATH/${1}_ec384_public.key | tee -a $CERTS_PATH/${1}_pub_keys.pem > /dev/null
+        cat $CERTS_PATH/${1}_rsa2048_public.key | tee -a $CERTS_PATH/${1}_pub_keys.pem > /dev/null
+
+        echo "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@" | \
+                tee $CERTS_PATH/${1}_keys.hash > /dev/null
+        echo "${1} key password is '${PASS_KEY}'" | tee -a $CERTS_PATH/${1}_keys.hash > /dev/null
```

Comment:
Create separate files for password and hashes .. that might be helpful for automation.

Reply:
Updated